---
description: Master the steps to import data from Phyloseq into MicrobiomeStat.
---

# Navigating Data from Phyloseq into MicrobiomeStat ğŸŒğŸ”„

Let's dive in! Your pathway from Phyloseq to MicrobiomeStat is as smooth as silk:

```r
# ğŸ“š Example code
# Don't forget to have the 'microbiome' package installed first!
library(microbiome)

# ğŸ“ Load the dataset
data(peerj32)
peerj32.phy <- peerj32$phyloseq

# ğŸ”„ Convert the Phyloseq object to a MicrobiomeStat data object
data.obj <- mStat_convert_phyloseq_to_data_obj(peerj32.phy)
```

Just like that, your Phyloseq object ğŸŒ¿ transmutes into a **MicrobiomeStat data object**.

With `mStat_convert_phyloseq_to_data_obj`, you acquire a **MicrobiomeStat data object** ğŸ’¾ (a list) brimming with:

* **feature.tab**: A matrix of the feature table ğŸ§¬ (OTU table). Superfluous rows with a sum of zero are swept away, leaving only the features present in the samples.
* **meta.dat**: A data frame of the sample data. This serves as a comprehensive ledger ğŸ“–, housing the metadata for each of your samples.
* **feature.ann**: A matrix of the feature annotation table ğŸ“œ (taxonomy table). Only the rows that have made their mark in the feature table are included.
* **tree**: A rooted phylogenetic tree ğŸŒ³. If not already rooted, the tree is given firm grounding by midpointing. Tips not present in the OTU table are pruned ğŸƒ.

By **weaving your Phyloseq object into the fabric of MicrobiomeStat format**, you awaken the boundless potential of MicrobiomeStat's analytical capabilities ğŸ”¬ and set the stage for your data to perform in the grand theatre of microbiome statistical analysis.

With your data object prepped and primed ğŸŒŸ, you're ready to embark on the captivating voyage of microbiome data analysis with MicrobiomeStat. Let the adventure begin! ğŸš€
